### PR TITLE
Make GitCoreRepository accept both main&git dir path from the outside instead of deriving the latter itself

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: frontendUiTable/src/main/java/com/virtuslab/gitmachete/frontend/ui/table/GitMacheteGraphTableManager.java [hunk 1]
```diff
@@ -110,12 +111,19 @@ private void refreshGraphTable(Path macheteFilePath, boolean isMacheteFilePresen
...
+    return Paths.get(gitRepository.getRoot().getPath());
+  }
+
+  private Path getGitDirectoryPath(GitRepository gitRepository) {
     // Using the deprecated `GitRepository#getGitDir` and not `GitRepository#getRepositoryFiles`
     // since the latter apparently doesn't allow to extract git dir
     // (just the paths of specific predefined files within the git dir).
-    Path gitDirPath = Paths.get(gitRepository.getGitDir().getPath());
-    return gitDirPath.resolve("machete");
+    return Paths.get(gitRepository.getGitDir().getPath());
```
Comment (micpiotrowski) [inline frontendUiTable/src/main/java/com/virtuslab/gitmachete/frontend/ui/table/GitMacheteGraphTableManager.java:122]: Same suggestion as on #209 :)

File: gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java [hunk 2]
```diff
@@ -39,58 +35,21 @@
...
-      Path parentDirectory = gitFilePath.getParent();
-      assert parentDirectory != null : "Can't get parent directory";
-
-      return parentDirectory.resolve(firstGroup).normalize();
-    }
-
-    throw new GitCoreNoSuchRepositoryException(
-        "File ${gitFilePath} does not contain a valid reference to .git directory");
-  }
-
```
Comment (micpiotrowski) [inline gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java:93]: bye-bey my sweet code  :wave: 
:upside_down_face:

Comment (PawelLipski) [reply gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java:93]: :cry:

File: frontendUiTable/src/main/java/com/virtuslab/gitmachete/frontend/ui/table/GitMacheteGraphTableManager.java (hunk 1 above)
Comment (PawelLipski) [reply frontendUiTable/src/main/java/com/virtuslab/gitmachete/frontend/ui/table/GitMacheteGraphTableManager.java:122]: Done, applied

File: gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java [hunk 3]
```diff
@@ -39,58 +35,21 @@
...
-    if (Files.isRegularFile(gitPath)) {
-      gitPath = getGitDirectoryPathFromGitFile(gitPath);
-    } else if (!Files.isDirectory(gitPath)) {
-      throw new GitCoreNoSuchRepositoryException("Git repository in path ${pathToRepoRoot} does not exist");
-    }
-
-    return gitPath;
-  }
-
-  private static Path getGitDirectoryPathFromGitFile(Path gitFilePath) throws GitCoreException {
```
Comment (mkondratek) [inline gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java:71]: <3

**Pull request merged**